app/scripts/views/mixins/open-webmail-mixin.js

```diff
@@ -0,0 +1,113 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+
+/**
+ * A view mixin that helps confirm views show a redirect button
+ * directly from FxA.
+ */
+define(function (require, exports, module) {
+  'use strict';
+
+  var BaseView = require('views/base');
+  var t = BaseView.t;
+  var _ = require ('underscore');
+
+  var WEBMAIL_SERVICES = [
+    {
+      buttonName: t('Open Gmail'),
+      link: 'https://mail.google.com/mail/u/?authuser=',
+      regex: /@gmail\.com$/,
+      webmailType: 'gmail'
+    },
+    {
+      buttonName: t('Open Hotmail'),
+      link: 'https://outlook.live.com/',
+      regex: /@hotmail\.com$/,
+      webmailType: 'hotmail'
+    },
+    {
+      buttonName: t('Open Yahoo'),
+      link: 'https://mail.yahoo.com',
+      regex: /@yahoo\.com$/,
+      webmailType: 'yahoo'
+    },
+    {
+      buttonName: t('Open Outlook'),
+      link: 'https://outlook.live.com/',
+      regex: /@outlook\.com$/,
+      webmailType: 'outlook'
+    },
+  ];
+
+  return {
+    events: {
+      'click #open-webmail': '_webmailTabOpened'
+    },
+
+    addUserInfo: function (providerLink, email) {
+
+      if (this.getWebmailType(email).contains('gmail')){
+        providerLink = providerLink.concat(encodeURIComponent(email));
+      }
+
+      return providerLink;
+    },
+
+    _getService: function (email) {
+      return _.find(WEBMAIL_SERVICES, function (service) {
+        return service.regex.test(email);
+      });
+    },
+
+    /**
+     * Monkey patch BaseView.prototype.getContext to return a context
+     * @method getContext
+     * @returns {Object}
+     */
+    getContext: function () {
+      var context = BaseView.prototype.getContext.call(this);
+      var email = context.email;
```

Comment:
Okay I checked it out and it seems like just for the sake of consistency it is not worth doing the `getAccount` bit. we can keep this just context.email, especially because we are calling the context right before. Could you just updated the conditional to be `if (email && context.openWebmailButtonVisible) {` to avoid calling webmail functions without an email.